src/graphql/direct.ts

```diff
@@ -38,76 +34,62 @@ export async function getDirectResources(): Promise<DirectEnvironment[]> {
-          id: connection.kafkaCluster.id,
-          name: "Kafka Cluster",
-          bootstrapServers: connection.kafkaCluster.bootstrapServers,
-          uri: connection.kafkaCluster.uri ?? "",
-          ...connectionInfo,
-        });
-      }
-
-      let schemaRegistry: DirectSchemaRegistry | undefined;
-      if (connection.schemaRegistry) {
-        schemaRegistry = DirectSchemaRegistry.create({
-          id: connection.schemaRegistry.id,
-          uri: connection.schemaRegistry.uri,
-          environmentId: connection.id as EnvironmentId,
-          ...connectionInfo,
-        });
-      }
+  const connection = response.directConnectionById;
+  if (!connection) {
+    return;
+  }
 
-      // combine the connection returned from GraphQL with the webview form augmented spec in storage
-      const directSpec: CustomConnectionSpec | undefined = directConnectionMap.get(
-        connection.id as ConnectionId,
-      );
+  const connectionInfo = {
+    connectionId: connection.id as ConnectionId,
+    connectionType: ConnectionType.Direct,
+  };
 
-      const directEnv = new DirectEnvironment({
-        id: connection.id as EnvironmentId,
-        name: connection.name,
-        kafkaClusters: kafkaCluster ? [kafkaCluster] : [],
-        kafkaConfigured: !!directSpec?.kafka_cluster,
-        schemaRegistry,
-        schemaRegistryConfigured: !!directSpec?.schema_registry,
-        formConnectionType: directSpec?.formConnectionType,
-        ...connectionInfo,
-      });
-      directResources.push(directEnv);
+  let kafkaCluster: DirectKafkaCluster | undefined;
+  if (connection.kafkaCluster) {
+    kafkaCluster = DirectKafkaCluster.create({
+      id: connection.kafkaCluster.id,
+      name: "Kafka Cluster",
+      bootstrapServers: connection.kafkaCluster.bootstrapServers,
+      uri: connection.kafkaCluster.uri ?? "",
+      ...connectionInfo,
     });
   }
 
-  // sort multiple environments by name
-  if (directResources.length > 1) {
-    directResources.sort((a, b) => a.name.localeCompare(b.name));
+  let schemaRegistry: DirectSchemaRegistry | undefined;
+  if (connection.schemaRegistry) {
+    schemaRegistry = DirectSchemaRegistry.create({
+      id: connection.schemaRegistry.id,
+      uri: connection.schemaRegistry.uri,
+      environmentId: connection.id as EnvironmentId,
+      ...connectionInfo,
+    });
   }
 
-  return directResources;
+  // look up the connectionId:spec map from storage
+  const directSpec: CustomConnectionSpec | null =
+    await getResourceManager().getDirectConnection(connectionId);
+
+  const env = new DirectEnvironment({
+    id: connection.id as EnvironmentId,
+    name: connection.name,
+    kafkaClusters: kafkaCluster ? [kafkaCluster] : [],
+    kafkaConfigured: !!directSpec?.kafka_cluster,
+    schemaRegistry,
+    schemaRegistryConfigured: !!directSpec?.schema_registry,
+    formConnectionType: directSpec?.formConnectionType,
+    ...connectionInfo,
+  });
+  env.isLoading = true;
```

Comment:
@jlrobins Special attention to this line -- while click-testing it looked like we didn't show any loading indicator after the initial rehydration, but maybe we don't need that here?

Reply:
Removed